cmd/main.go

```diff
@@ -0,0 +1,48 @@
+package main
+
+import (
+	api "github.com/ChainSafe/gossamer/internal"
+	"os"
+
+	cfg "github.com/ChainSafe/gossamer/config"
+	"github.com/ChainSafe/gossamer/p2p"
+	"github.com/ChainSafe/gossamer/polkadb"
+	"github.com/ChainSafe/gossamer/rpc"
+	"github.com/ChainSafe/gossamer/rpc/json2"
+	log "github.com/inconshreveable/log15"
+	"github.com/naoina/toml"
+)
+
+func main() {
```

Comment:
This is an important point, although I can modify this into isolated functions I am now thinking that the entrypoint PR should come after the CLI mainly because of how much will change here and since the entrypoint isn't really doing anything right now it won't hinder our development if we postpone until an initial CLI is completed.